src/app/producer-shelf/components/producer-shelf/producer-shelf.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, ElementRef, HostListener, OnDestroy, ViewChild } from '@angular/core'
+import { KeyboardBindingService } from '../../abstractions/keyboard-binding.service'
+import { KeyBinding } from '../../models/key-binding'
+import { KeyboardBindingMatcher } from '../../services/keyboard-binding.matcher'
+import { IconButton, IconButtonSize } from '../../../shared/enums/icon-button'
+
+@Component({
+  selector: 'sofie-producer-shelf',
+  templateUrl: './producer-shelf.component.html',
+  styleUrls: ['./producer-shelf.component.scss'],
+})
+export class ProducerShelfComponent implements OnDestroy {
```

Comment:
What's the reason for calling it "producer" shelf?

Reply:
I extracted the drag/resize functionality into a wrapper component named `DraggableShelf`.